src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabAuthorizationInterceptor.java

```diff
@@ -22,6 +22,11 @@ public class GitLabAuthorizationInterceptor implements ClientHttpRequestIntercep
     @NotNull
     @Override
     public ClientHttpResponse intercept(HttpRequest request, @NotNull byte[] body, @NotNull ClientHttpRequestExecution execution) throws IOException {
+        return intercept(request, body, execution, gitlabPrivateToken);
+    }
+
+    public static ClientHttpResponse intercept(HttpRequest request, @NotNull byte[] body, @NotNull ClientHttpRequestExecution execution, String gitlabPrivateToken)
```

Comment:
I think it would be better here to not have this additional public method that passes in the private key even if that would lead to a small duplication of the two lines in `GitlabCIAuthorizationInterceptor`.

You could maybe define a public static string constant here with the header name and use that one in the other class. That way the duplication is kept to a minimum and a possible future change of the header name has to be applied once in the code.